owncloudComLibrary/src/main/java/com/owncloud/android/lib/common/authentication/oauth/OAuthConnectionBuilder.kt

```diff
@@ -0,0 +1,31 @@
+package com.owncloud.android.lib.common.authentication.oauth
+
+import android.net.Uri
+import net.openid.appauth.Preconditions
+import net.openid.appauth.connectivity.ConnectionBuilder
+import java.io.IOException
+import java.net.HttpURLConnection
+import java.net.URL
+import java.util.concurrent.TimeUnit
+
+/**
+ * Based on [net.openid.appauth.connectivity.DefaultConnectionBuilder] but permitting http connections in addition
+ * to https connections
+ */
+class OAuthConnectionBuilder : ConnectionBuilder {
+    @Throws(IOException::class)
+    override fun openConnection(uri: Uri): HttpURLConnection {
+        Preconditions.checkNotNull(uri, "url must not be null")
```

Comment:
I wouldn't add that check not null since if uri is null, kotlin will throw an exception. I don't know if that method performs aditional checks, so lets move this forward.